Sources/Playlist/KPPlaylistController.swift

```diff
@@ -324,8 +326,10 @@ import PlayKit
                     }
                 }
                 
-                let mediaOptions = self.prepareMediaOptions(forMediaEntry: mediaEntry)
-                self.player?.setMediaAndUpdatePlugins(mediaEntry: mediaEntry, mediaOptions: mediaOptions, pluginConfig: pluginConfig, callback: { error in
+                self.player?.setMediaAndUpdatePlugins(mediaEntry: currentEntry,
+                                                      mediaOptions: mediaOptions,
+                                                      pluginConfig: pluginConfig,
+                                                      callback: { error in
                     self.messageBus?.post(PlaylistEvent.PlaylistCurrentPlayingItemChanged())
                 })
```

Comment:
Same here, missing weak self in block.

```suggestion
                self.player?.setMediaAndUpdatePlugins(mediaEntry: currentEntry,
                                                      mediaOptions: mediaOptions,
                                                      pluginConfig: pluginConfig,
                                                      callback: { [weak self] error in
                    self?.messageBus?.post(PlaylistEvent.PlaylistCurrentPlayingItemChanged())
                })
```